src/modules/datasets/ingestion_parquet.bicep

```diff
@@ -0,0 +1,60 @@
+// Description: An Azure Data Factory dataset for parquet files in Azure Blob Storage
+
+@description('Required. The name of the parent Azure Data Factory.')
+param dataFactoryName string
+
+@description('Required. The storage account where the data resides.')
+param linkedServiceName string
+
+@description('Required. Name.')
+param datasetName string
+
+var datasetType = 'Parquet'
+var locationType = 'AzureBlobFSLocation'
+var compressionCodec = 'gzip'
+
+resource dataFactoryRef 'Microsoft.DataFactory/factories@2018-06-01' existing = {
+  name: dataFactoryName
+}
+
+resource dataset 'Microsoft.DataFactory/factories/datasets@2018-06-01' =  {
+  name: datasetName
+  parent: dataFactoryRef
+  properties: {
+    annotations: []
+    parameters: {
+      fileName: {
+        type: 'String'
+      }
+      folderName: {
+        type: 'String'
+      }
+    }
+    type: datasetType
+    typeProperties: {
```

Comment:
Given this is still 90% the same as the other datasets, I'd merge this into one dataset.bicep and change the properties based on the datasetType. You can do something like:
```bicep
var csvProps = { ... }
var parquetProps = { ... }
var commonProps = { ... }
...
typeProperties: union(commonProps, datasetType == 'Parquet' ? parquetProps : csvProps)
```